n2y/plugins/footnotes.py

```diff
@@ -0,0 +1,58 @@
+import re
+
+from pandoc.types import Note, Str, Para
+
+from n2y.rich_text import TextRichText
+from n2y.blocks import ParagraphBlock
+
+
+class ParagraphWithFootnoteBlock(ParagraphBlock):
+    def __init__(self, client, notion_data, page, get_children=True):
+        super().__init__(client, notion_data, page, get_children)
+        self._attach_footnote_data_if_exists()
+
+    def to_pandoc(self):
+        return None if self._is_footnote() else super().to_pandoc()
+
+    def _attach_footnote_data_if_exists(self):
+        if self._is_footnote():
+            if "footnotes" not in self.client.plugin_data:
+                self.client.plugin_data["footnotes"] = {}
+            self.client.plugin_data["footnotes"][self._footnote()] = self._footnote_ast()
```

Comment:
I'd show a warning if the same footnote shows up twice in the same file (and be sure to include a link to the second footnote block)